src/v2/services/implementations/AssetsService.ts

```diff
@@ -26,33 +28,51 @@ export class AssetsService implements IAssetsService {
   }
 
   public async transferNativeAsset(param: ParamAssetTransfer): Promise<void> {
-    const transaction = await this.AssetsRepository.getNativeTransferCall(param);
-    const hash = await this.wallet.signAndSend({
-      extrinsic: transaction,
-      senderAddress: param.senderAddress,
-      successMessage: param.successMessage,
-    });
-    param.finalizedCallback(String(hash));
+    const useableBalance = await this.AssetsRepository.getNativeBalance(param.senderAddress);
+    const isBalanceNotEnough =
+      Number(ethers.utils.formatUnits(useableBalance, 18)) -
```

Comment:
In `useChainMetadata` you can find `decimals` value, or do what @impelcrypto suggested to avoid passing number of decimals

Reply:
@bobo-k2 
Use `ethers.utils.formatEther(wei)`, so that no need to pass decimals.
https://github.com/AstarNetwork/astar-apps/pull/1116#discussion_r1448929113

Reply:
Even better, @gluneau I apologize for confusion